lib/document.js

```diff
@@ -987,7 +987,8 @@ Document.prototype.$session = function $session(session) {
 
   this.$__.session = session;
 
-  if (!this.ownerDocument) {
+
+  if (!this.isSubDocument) {
```

Comment:
Use `this.$__.isSubDocument` instead

Reply:
Just there or all the other instances?

Reply:
I ran it with only replacing that and replacing the other instances and still get a bunch of test failures